app/src/main/java/com/infomaniak/swisstransfer/workers/UploadWorker.kt

```diff
@@ -0,0 +1,143 @@
+            return Result.failure()
+        }
+
+        val uploadSession = uploadManager.initUploadSession(recaptcha = "Recaptcha")!!
+        val totalSize = uploadSession.files.sumOf { it.size }
+
+        SentryLog.d(TAG, "Work started with ${uploadSession.files.count()} files of $totalSize bytes")
+
+        uploadSession.files.forEach { fileSession ->
+            uploadFileTask.start(fileSession, uploadSession) { bytesSent ->
+                totalUploadedBytes += bytesSent
+                setProgress(workDataOf(UPLOADED_BYTES_TAG to totalUploadedBytes, TOTAL_SIZE_TAG to totalSize))
+            }
+        }
+
+        uploadManager.finishUploadSession(uploadSession.uuid)
+        importLocalStorage.removeImportFolder()
+
+        return Result.success()
+    }
+
+    override fun onFinish() {
+        SentryLog.i(TAG, "Work finished")
+    }
+
+    @Singleton
+    class Scheduler @Inject constructor(private val workManager: WorkManager) {
+
+        fun scheduleWork() {
+            SentryLog.i(TAG, "Work scheduled")
+            val workRequest = OneTimeWorkRequestBuilder<UploadWorker>()
+                .setConstraints(Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build())
+                .build()
+            workManager.enqueueUniqueWork(TAG, ExistingWorkPolicy.REPLACE, workRequest)
+        }
+
+        @OptIn(ExperimentalCoroutinesApi::class)
+        fun trackUploadProgressFlow(): Flow<UploadTransferProgress> {
+            val workQuery = WorkQuery.Builder.fromUniqueWorkNames(listOf(TAG))
+                .addStates(listOf(WorkInfo.State.RUNNING))
+                .build()
+            return workManager.getWorkInfosFlow(workQuery).mapLatest {
+                val workProgress = it.firstOrNull()?.progress ?: return@mapLatest null
+                UploadTransferProgress(
+                    uploadedSize = workProgress.getLong(UPLOADED_BYTES_TAG, 0L),
+                    totalSize = workProgress.getLong(TOTAL_SIZE_TAG, 0L),
+                )
+            }.filterNotNull()
+        }
+
+        @OptIn(ExperimentalCoroutinesApi::class)
+        fun isPendingOrRunningFlow(): Flow<Boolean> {
+            val workQuery = WorkQuery.Builder.fromUniqueWorkNames(listOf(TAG))
+                .addStates(listOf(WorkInfo.State.RUNNING, WorkInfo.State.ENQUEUED, WorkInfo.State.BLOCKED))
+                .build()
+            return workManager.getWorkInfosFlow(workQuery).mapLatest { it.isNotEmpty() }
+        }
+
+        fun cancelWork() {
+            SentryLog.i(TAG, "Work cancelled")
+            workManager.cancelUniqueWork(TAG)
+        }
+    }
+
+    @Immutable
+    data class UploadTransferProgress(
+        val uploadedSize: Long,
+        val totalSize: Long,
+    )
+
+    companion object {
+        private const val TAG = "UploadWorker"
+        private const val EXPECTED_CHUNK_SIZE = 50L * 1024 * 1024 // 50Mo
+        private const val TOTAL_FILE_SIZE = 50L * 1024 * 1024 * 1024  // 50Go
```

Comment:
We have the same constant here and in FileCHunkSizeManager. It might be better to use the same one everywhere and only have one constant for each ?